crates/rpc-types/src/eth/admin.rs

```diff
@@ -0,0 +1,410 @@
+//! Types for the admin api
+use alloy_genesis::ChainConfig;
+use alloy_primitives::{B256, U256};
+use alloy_serde::json_u256::deserialize_json_u256;
+use enr::{k256::ecdsa::SigningKey, Enr};
+use serde::{Deserialize, Serialize};
+use std::net::{IpAddr, SocketAddr};
+
+/// This includes general information about a running node, spanning networking and protocol
+/// details.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct NodeInfo {
```

Comment:
this can be extracted from reth,

including the NodeRecord type

https://github.com/paradigmxyz/reth/blob/4d798c7b655255b84cbbcd3af2e8827de81e105e/crates/rpc/rpc-types/src/admin.rs#L16-L16

Reply:
`NodeRecord` includes a lot of logic to parse `enode`, which I remember can't handle edge case like missing ports.